src/app/clusters/ethernet-network-diagnostics-server/ethernet-diagnostics-cluster.h

```diff
@@ -0,0 +1,144 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/clusters/ethernet-network-diagnostics-server/ethernet-diagnostics-logic.h>
+
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <clusters/EthernetNetworkDiagnostics/ClusterId.h>
+#include <clusters/EthernetNetworkDiagnostics/Commands.h>
+#include <clusters/EthernetNetworkDiagnostics/Events.h>
+#include <clusters/EthernetNetworkDiagnostics/Metadata.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/support/Span.h>
+#include <protocols/interaction_model/StatusCode.h>
+#include <sys/types.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+/// Integration of Ethernet network diagnostics logic within the Matter data model
+///
+/// Translates between matter calls and Ethernet Network Diagnostics logic
+///
+/// This cluster is expected to only ever exist on endpoint 0 as it is a singleton cluster.
+template <typename LOGIC>
+class EthernetDiagnosticsServerCluster : public DefaultServerCluster, private LOGIC
+{
+public:
+    template <typename... Args>
+    EthernetDiagnosticsServerCluster(Args &&... args) :
+        DefaultServerCluster({ kRootEndpointId, EthernetNetworkDiagnostics::Id }), LOGIC(std::forward<Args>(args)...)
+    {}
+
+    // Server cluster implementation
+    DataModel::ActionReturnStatus ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                AttributeValueEncoder & encoder) override
+    {
+        switch (request.path.mAttributeId)
+        {
+        case EthernetNetworkDiagnostics::Attributes::PHYRate::Id:
+            return LOGIC::ReadPHYRate(encoder);
+        case EthernetNetworkDiagnostics::Attributes::FullDuplex::Id:
+            return LOGIC::ReadFullDuplex(encoder);
+        case EthernetNetworkDiagnostics::Attributes::CarrierDetect::Id:
+            return LOGIC::ReadCarrierDetect(encoder);
+        case EthernetNetworkDiagnostics::Attributes::TimeSinceReset::Id: {
+            uint64_t value;
+            CHIP_ERROR err = LOGIC::GetTimeSinceReset(value);
+            return EncodeValue(value, err, encoder);
+        }
+        case EthernetNetworkDiagnostics::Attributes::PacketRxCount::Id: {
+            uint64_t value;
+            CHIP_ERROR err = LOGIC::GetPacketRxCount(value);
```

Comment:
Since all the attributes are optional, can we check `mEnabledAttributes.enablePacketRxCount` before calling `LOGIC`?

Reply:
This problem is for all migrated clusters.

Reply:
What we need is `Attributes` to return the right list of attributes. I think we are somewhat unclear on what `dynamic data model` definition is. 

My understanding is that we have:
  - static/fixed clusters in ember - this is obvious
  - dynamic clusters in ember, these currently use a `some (maybe disabled) endpoint MUST enable all required attributes` then you can use dynamic endpoints. I believe this is what you use, but "some disabled endpoint" does not work well with singletongs. Need to understand what you actually enable in zap today
  - dynamic/code-driven app will not use CodegenIntegration at all, will have code that instantiates the clusters instead.

Reply:
Adding an enable check for each attribute read and write would further increase flash usage, which is the main blocker for this transition.

I propose deferring the new logic to a follow-up PR, so we can first evaluate the flash and RAM impact of this change.

If an attribute is not enabled, all read and write requests should be blocked at the IM layer before reaching the cluster server.

I don’t believe ServerClusterInterface currently supports dynamic endpoints. I’d prefer to have a separate discussion on that topic. It doesn’t apply to this cluster anyway, as it’s always enalbed on the root endpoint.